components/action-mgt/org.wso2.carbon.identity.action.management/src/main/java/org/wso2/carbon/identity/action/management/dao/impl/ActionManagementDAOFacade.java

```diff
@@ -294,6 +307,108 @@ private void addEncryptedAuthSecretsToBuilder(ActionDTOBuilder actionDTOBuilder,
                         .build());
     }
 
+    private void addActionRule(ActionDTOBuilder actionDTOBuilder, String tenantDomain) throws ActionMgtException {
+
+        if (actionDTOBuilder.getActionRule() == null || actionDTOBuilder.getActionRule().getRule() == null) {
```

Comment:
Shall we remove these if condition as these are checked in the `isAddingNewActionRule()` method ?

Reply:
Well this is used in addAction not just update. So we can't remove that from here. Taking this out also impacts readability. So let's keep it.